### PR TITLE
Removes all IDS-REST modules and related code

### LINKED ISSUE #359: Delete IDS Sample Code
<sub>Dominik Pinsel <dominik.pinsel@daimler.com>, Daimler TSS GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md) </sub>

### PR DESCRIPTION
**Pull request by paullatzelsperger**
This PR removes all java modules related to IDS REST and related code, most notably:
- ` ids-api-transfer`: previously used for initiating data transfers. Superseded by the Contract Negotiation + Transfer via IDS Multipart.
- `ids-api-catalog`: previously used for querying a connector's catalog. Superseded by the Federated Catalog.
- `DataRequestMessageSender` and `QueryMessageSender`: related message senders for the aforementioned endpoints.
- `PolicyRegistry`: removed as policies are now attached to `ContractOffer` objects.

_Note: Our "numbered" Samples have been checked and updated, however all samples in the `other` folder have not and might not work anymore. Respective developers have been notified and absent an action on their part those samples will get removed subsequently!_

Closes #359

### CODE REVIEW

File: samples/04-file-transfer/transfer-file/src/main/java/org/eclipse/dataspaceconnector/extensions/api/FileTransferExtension.java [hunk 1]
```diff
@@ -61,28 +58,25 @@ public void initialize(ServiceExtensionContext context) {
         DataFlowController dataFlowController = new InlineDataFlowController(vault, context.getMonitor(), dataOperatorRegistry, dataAddressResolver);
         dataFlowMgr.register(dataFlowController);
 
-        var policy = savePolicies(context);
```
Comment (ronjaquensel) [inline samples/04-file-transfer/transfer-file/src/main/java/org/eclipse/dataspaceconnector/extensions/api/FileTransferExtension.java:64]: Just a minor thing, but I think this should be adjusted in the sample's README as well.

**Review by ronjaquensel — APPROVED** [review]

**Review by ndr-brt — APPROVED** [review]

File: samples/other/file-transfer-s3-to-s3/src/main/java/org/eclipse/dataspaceconnector/transfer/demo/api/ClientApiController.java [hunk 2]
```diff
@@ -115,9 +88,13 @@ private DataRequest createRequest(
             String destinationRegion
     ) {
 
+        if (1 == 1) {
```
Comment (alexandrudanciu) [inline samples/other/file-transfer-s3-to-s3/src/main/java/org/eclipse/dataspaceconnector/transfer/demo/api/ClientApiController.java:91]: If this always throws an exception, why not deleting the whole module?

**Review by alexandrudanciu — COMMENTED** [review]
Review resulted in on question of understanding.

File: samples/other/file-transfer-s3-to-s3/src/main/java/org/eclipse/dataspaceconnector/transfer/demo/api/ClientApiController.java (hunk 2 above)
Comment (paullatzelsperger) [reply samples/other/file-transfer-s3-to-s3/src/main/java/org/eclipse/dataspaceconnector/transfer/demo/api/ClientApiController.java:91]: oops, I put this in during debugging and never deleted it :/

**Review by alexandrudanciu — APPROVED** [review]
Review comment was addressed.

File: samples/04-file-transfer/transfer-file/src/main/java/org/eclipse/dataspaceconnector/extensions/api/FileTransferExtension.java (hunk 1 above)
Comment (paullatzelsperger) [reply samples/04-file-transfer/transfer-file/src/main/java/org/eclipse/dataspaceconnector/extensions/api/FileTransferExtension.java:64]: done. also renamed the method to `createPolicy`

**Pull request merged**